wcfsetup/install/files/js/WoltLabSuite/Core/ImageResizer.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * This module allows resizing and conversion of HTMLImageElements to Blob and File objects
+ *
+ * @author	Maximilian Mader
+ * @copyright	2001-2018 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module	WoltLabSuite/Core/ImageResizer
+ */
+define([
+	'WoltLabSuite/Core/FileUtil',
+	'WoltLabSuite/Core/ExifUtil',
+	'Pica'
+], function(FileUtil, ExifUtil, Pica) {
+	var pica = Pica({ features: [ 'js', 'wasm', 'ww' ] });
+	
+	/**
+	 * @constructor
+	 */
+	function ImageResizer() { }
+	
+	ImageResizer.prototype = {
+		maxWidth: 800,
+		maxHeight: 600,
+		quality: 0.8,
+		fileType: 'image/jpeg',
+		
+		setMaxWidth: function (value) {
+			if (value == null) value = ImageResizer.prototype.maxWidth;
+			
+			this.maxWidth = value;
+			return this;
+		},
+		
+		setMaxHeight: function (value) {
+			if (value == null) value = ImageResizer.prototype.maxHeight;
+			
+			this.maxHeight = value;
+			return this;
+		},
+		
+		setQuality: function (value) {
+			if (value == null) value = ImageResizer.prototype.quality;
+			
+			this.quality = value;
+			return this;
+		},
+		
+		setFileType: function (value) {
+			if (value == null) value = ImageResizer.prototype.fileType;
+			
+			this.fileType = value;
+			return this;
+		},
+		
+		getFile: function (result, fileName, fileType, quality) {
+			var image = result.image;
+			var exif = result.exif;
+			
+			return pica.toBlob(image, fileType, quality)
+				.then(function (blob) {
+					if (fileType === 'image/jpeg' && typeof exif !== 'undefined') {
+						blob = ExifUtil.setExifData(blob, exif);
+					}
+					
+					var matches = fileName.match(/(.+)(\..+?)$/);
+					
+					// If we inserted EXIF data blob has been converted to a Promise
+					return Promise.resolve(blob)
+						.then(function (blob) {
```

Comment:
- Append this `.then` to the outer Promise
- Move `var matches` into the `.then`.
- You now can `return ExifUtil.setExifData(blob, exif)` in the `if` above and `return blob` otherwise.